include/gatekeeper_lls.h

```diff
@@ -211,6 +211,16 @@ struct lls_config {
 };
 
 /*
+ * TODO Enable an lcore to subscribe multiple times for the same IP address.
+ * This is needed because it's very hard for the GK blocks avoid duplicates.
+ * Currently, the LLS block has trouble with unsubscribing those duplicates
+ * since the LLS block identifies what to remove using
+ * the pair <IP_address, lcore_id>.
+ * In addition, the LLS block might not accept a subscription because
+ * it saves memory for a single subscription for lcore id.
+ */
```

Comment:
Originally, it was a conscious design decision to push the task of avoiding duplicates to the individual blocks, instead of putting it in the LLS cache. For example, the CPS block maintains a linked list of ARP and ND holds that it has made so that it does not issue duplicate requests.

We can revisit that decision now, if needed, but I think it should be possible for the GK block to keep track of duplicates. Either way, I think we should handle this issue in this patch (and add the `hold_{arp,nd}` and `put_{arp,nd}` and callback functions) so that we can see both the writing and reading sides of the seqlocks added here.